### DISCUSSION TITLE
关于tool_calls和function_call的问题

### DISCUSSION THREAD

**Discussion by kylinccc** [zai-org/ChatGLM3#1246] [Q&A] [answered]
我在调用API平台的GLM4和GLM3 turbo的工具模式时,返回的response里的信息显示使用的tool_calls,
在本地部署的chatglm3-6b的工具模式下,返回的response里的信息显示使用的function_call
请问一下这两者有什么不同吗,是本来就这样还是我配置不对

**Comment by kylinccc** [comment]
![WeChat0cbe55f402fe2c248565d7c8910f53a5](https://github.com/THUDM/ChatGLM3/assets/62291299/89a1db63-05df-44ba-84ce-2df18c88808b)
![WeChata0e60ea0d9adaf0402b08b9f56f4dc7c](https://github.com/THUDM/ChatGLM3/assets/62291299/b0cab951-560e-4c5b-ab83-f185a5a4fbc4)

**Comment by kylinccc** [comment]
![WeChatfcc76d084cbe890173cdbfca5e751f31](https://github.com/THUDM/ChatGLM3/assets/62291299/53730fa2-4107-488d-a53f-fad0a6c8c8ea)
只有此处代码不同,其他都是一样的

**Comment by zRzRzRzRzRzRzR** [comment, marked as answer]
嗯，是的，之后我们会对齐这部分的写法，根openai一样

**Reply by kylinccc** [reply]
明白了,感谢回应和解答
